src/js/components/Calendar/Calendar.js

```diff
@@ -589,23 +590,36 @@ const Calendar = forwardRef(
       return (
         <Box direction="row" justify="between" align="center">
           <Box flex pad={{ horizontal: headingPadMap[size] || 'small' }}>
-            <Heading
-              level={
-                size === 'small'
-                  ? (theme.calendar.heading && theme.calendar.heading.level) ||
-                    4
-                  : ((theme.calendar.heading && theme.calendar.heading.level) ||
-                      4) - 1
-              }
-              size={size}
-              margin="none"
-              overflowWrap="normal"
-            >
-              {reference.toLocaleDateString(locale, {
-                month: 'long',
-                year: 'numeric',
-              })}
-            </Heading>
+            {theme.calendar.heading && theme.calendar[size].titleSize ? (
```

Comment:
Why do we need the check for `theme.calendar.heading`?

Reply:
Removed this extra check in latest commit